moabb/evaluations/splitters.py

```diff
@@ -101,3 +111,138 @@ def split(self, y, metadata):
+    for dividing the data into training and testing sets based on sessions.
+
+    .. image:: images/crosssess.jpg
+        :alt: The schematic diagram of the CrossSession split
+        :align: center
+
+    The inner cross-validation strategy can be changed by passing the
+    `cv_class` and `cv_kwargs` arguments. By default, it uses LeaveOneGroupOut,
+    which effectively performs Leave-One-Session-Out cross-validation when
+    sessions are passed as groups.
+
+    Parameters
+    ----------
+    cv_class: cross-validation class, default=LeaveOneGroupOut
+        Inner cross-validation strategy for splitting the sessions.
+        For cross-session splitting, LeaveOneGroupOut is the most suitable as default.
+    shuffle: bool, default=False
+        Whether to shuffle the session order for each subject. By default, it is not
+        used because of LeaveOneGroupOut's determinist behaviour.
+    random_state: int, RandomState instance or None, default=None
+        Controls the randomness when `shuffle` is True.
+        Pass an int for reproducible output across multiple function calls.
+        Be default, it is not used because of the determinist behaviour of
+        LeaveOneGroupOut.
+    cv_kwargs: dict
+        Additional arguments to pass to the inner cross-validation strategy.
+
+    Yields
+    ------
+    train : ndarray
+        The training set indices for that split.
+
+    test : ndarray
+        The testing set indices for that split.
+    """
+
+    def __init__(
+        self,
+        cv_class: type[BaseCrossValidator] = LeaveOneGroupOut,
+        shuffle: bool = False,
+        random_state: int = None,
+        **cv_kwargs,
+    ):
+        self.cv_class = cv_class
+        self.cv_kwargs = cv_kwargs
+        self._cv_kwargs = dict(**cv_kwargs)
+
+        self.shuffle = shuffle
+        self.random_state = random_state
+
+        self._rng = check_random_state(random_state) if shuffle else None
+
+        if not shuffle and random_state is not None:
+            raise ValueError("`random_state` should be None when `shuffle` is False")
+
+        if shuffle and len(self.cv_kwargs) == 0 and cv_class is LeaveOneGroupOut:
+            log.warning(
+                "Shuffling is not implemented for LeaveOneGroupOut. "
+                "The `shuffle` parameter change the behaviour of the splitter."
+                "Use GroupShuffleSplit instead."
+            )
+
+            self.cv_class = GroupShuffleSplit
+
+        params = inspect.signature(self.cv_class).parameters
+        for p, v in [
+            ("shuffle", shuffle),
+            ("random_state", self._rng),
+        ]:
+            if p in params:
+                self._cv_kwargs[p] = v
+
+    def get_n_splits(self, metadata):
+        if self.cv_class is LeaveOneGroupOut:
+            return metadata.groupby(["subject", "session"]).ngroups
+        else:
+            return self.cv_class(**self._cv_kwargs).get_n_splits(metadata)
```

Comment:
I don't understand this function. Why not something like this:
```suggestion
        subjects = metadata["subject"].unique()
        n_splits = 0
        for subject in subjects:
             sessions = subject_metadata["session"].unique()
             if len(sessions) <= 1:
                 continue  # Skip subjects with only one session
             splitter = self.cv_class(**self._cv_kwargs)
             n_splits += splitter.get_n_splits(subject_metadata)
```

Reply:
I think the idea here is not consume the splitter before necessary.

Reply:
Then could you add a docstring regarding what the function should return? Because it's not super clear to me..

Reply:
I agree that the implementation by pierre seems more true no?

Reply:
yessss